Correct name of WW3 JRA test

sbrus89 opened the pull request:
Correct an error in the compset name for the WW3 extended coverage test.  Reduce the amount of data downloaded.

[BFB]  new test

rljacob commented:
merged to next.

sbrus89 commented:
Thanks @rljacob!

rljacob commented:
Finally looked at this running and there's a problem:
https://my.cdash.org/test/68672096
There are permission problems on that machine but the issue I'm concerned about is that its downloading to much data.   If its only running 3 timesteps, it shouldn't download mutli years of input data.

ndkeen commented:
Noting https://github.com/E3SM-Project/E3SM/issues/5349

rljacob commented:
And https://github.com/E3SM-Project/E3SM/pull/5150 showed how to handle this for one case.

rljacob commented:
@bishtgautam can you advise?  This test is still trying to download multiple years of data.

jonbob commented:
@rljacob - I think the changes @bishtgautam made to the datm forcing so the JRA set would use the DATM_CLMNCEP_YR_START and DATM_CLMNCEP_YR_STOP settings was only for "JRA" and not for "JRA1p4". At the time those changes were made, we discussed a more general approach and I'm intending to take that on under Issue #5349. But for now, the compset would have to be modified to use "JRA" instead.

bishtgautam commented:
@rljacob Here are the changes needed to look for only 2004 settings.

```
diff --git a/cime b/cime
--- a/cime
+++ b/cime
@@ -1 +1 @@
-Subproject commit 977c1f6cd9283d03e356c8370a32302be4991cf4
+Subproject commit 977c1f6cd9283d03e356c8370a32302be4991cf4-dirty
diff --git a/components/data_comps/datm/cime_config/namelist_definition_datm.xml b/components/data_comps/datm/cime_config/namelist_definition_datm.xml
index e3dcbf4..209b812 100644
--- a/components/data_comps/datm/cime_config/namelist_definition_datm.xml
+++ b/components/data_comps/datm/cime_config/namelist_definition_datm.xml
@@ -1813,6 +1813,7 @@
       <value stream="CORE2_IAF.NCEP.U_10">1948</value>
       <value stream="CORE2_IAF.NCEP.V_10">1948</value>
       <value stream="CORE2_IAF.CORE2.ArcFactor">1948</value>
+      <value stream="CORE_IAF_JRA.*2018">$DATM_CLMNCEP_YR_START</value>
       <value stream="CORE_IAF_JRA.*">$DATM_CLMNCEP_YR_START</value>
       <value stream="CORE_RYF8485_JRA">1984</value>
       <value stream="CORE_RYF9091_JRA">1990</value>
@@ -1885,7 +1886,7 @@
       <value stream="CORE2_IAF.CORE2.ArcFactor">2009</value>
       <value stream="co2tseries.20tr" cime_model="e3sm">2007</value>
       <value stream="co2tseries.20tr" cime_model="cesm">2014</value>
-      <value stream="CORE_IAF_JRA.*2018">2018</value>
+      <value stream="CORE_IAF_JRA.*2018">$DATM_CLMNCEP_YR_END</value>
       <value stream="CORE_IAF_JRA.*">$DATM_CLMNCEP_YR_END</value>
       <value stream="CORE_RYF8485_JRA">1984</value>
       <value stream="CORE_RYF9091_JRA">1990</value>
```

Btw, I think all entries should be using `$DATM_CLMNCEP_YR_START` and `$DATM_CLMNCEP_YR_END` values.

jonbob commented:
@rljacob -- I agree with @bishtgautam that all entries should be using something to set the start and stop year. When we made the PR to change this for the "JRA" forcing, we discussed renaming DATM_CLMNCEP_* to something more general, which I would prefer

rljacob commented:
Thanks @bishtgautam !  Yes we should change "CLMNCEP" to maybe "FORCING".  for another PR.

rljacob commented:
@sbrus89 can you add the changes discussed above to this branch?

sbrus89 commented:
@rljacob - yes, will do.

rljacob commented:
merged to next

rljacob commented:
@sbrus89 this has a new error: ERROR: No description found for comp_class atm matching compsetname 2000_DATM%JRA-2018_SLND_MPASSI_MPASO%DATMFORCED_DROF%JRA-2018_SGLC_WW3%sp36x36_SIAC_SESP in file /home/jason_sarich/E3SM/components/data_comps/datm/cime_config/config_component.xml, expected match in ['DATM'] % ['QIA', 'WISOQIA', 'CRU', 'CRUv7', 'GSWP3v1', 'MOSARTTEST', 'NLDAS2', 'CPLHIST', '1PT', 'NYF', 'IAF', 'JRA', 'JRA-1p5', 'JRA-1p4-2018', 'JRA-RYF8485', 'JRA-RYF9091', 'JRA-RYF0304', 'CFSv2', 'CFSR']

darincomeau commented on components/ww3/cime_config/config_compsets.xml:
```diff
@@ -30,8 +30,8 @@
-    <alias>GMPAS-JRA1p4-WW3</alias>
-    <lname>2000_DATM%JRA-1p4-2018_SLND_MPASSI_MPASO%DATMFORCED_DROF%JRA-1p4-2018_SGLC_WW3%sp36x36</lname>
+    <alias>GMPAS-JRA-WW3</alias>
+    <lname>2000_DATM%JRA-2018_SLND_MPASSI_MPASO%DATMFORCED_DROF%JRA-2018_SGLC_WW3%sp36x36</lname>
```
@sbrus89 I think you want `JRA-2018` -> `JRA-1p5` in the two instances here for JRA v1.5. If you did want JRA v1.4 instead, than the previous `JRA-1p4-2018` is the correct one.

sbrus89 commented:
This relies on #5639  for the relevant `$DATM_CLMNCEP_YR_END` and `$DATM_CLMNCEP_YR_START` changes to `IAF_JRA_1p5.*`

rljacob commented:
merged to next.

jonbob commented:
You may want to add the settings from [PR #5670](https://github.com/E3SM-Project/E3SM/pull/5670) since this compset appears to use DROF

rljacob commented:
@sbrus89 can you use any year?  The other test with JRA1p5 forcing uses 1958.  The tests that use this forcing should settle on the particular year.

sbrus89 commented:
@rljacob - yes, I can change to 1958.
@jonbob - I'll add the shell commands:
```
./xmlchange DROF_STRM_YR_START=1958
./xmlchange DROF_STRM_YR_END=1958
```

jonbob commented:
@sbrus89 and @rljacob -- I just picked the first year of the forcing in the D-case test for no real reason, other than it wouldn't change answers. Please feel free to change that one to 2004 as well, if there's a reason you want to use that year

jonbob commented:
@sbrus89 -- can we also add these commands to the testdef?
````
./xmlchange DROF_STRM_YR_START=2004
./xmlchange DROF_STRM_YR_END=2004
````
This is still not passing on gcp12 because it's trying to download all the drof files. I'm also happy to make this change and push to your branch, if that's easier...

sbrus89 commented:
@jonbob, its probably faster if you just push to my branch.

jonbob commented:
@sbrus89 -- turns out they were already there. I think this just needs to get merged to next again. @rljacob?

rljacob commented:
What year did we want this test to run with?  2004 or 1958?

rljacob commented:
I think 1958 since this JRA test also uses that year:  SMS_D_Ld1.TL319_EC30to60E2r2.DTESTM-JRA1p5.pm-cpu_gnu.mpassi-jra_1958

jonbob commented:
I think as-is this branch has 1958 for everything, so it should be good to go. I was just referencing 2004 because the test on gcp12 has that date, but that's due to an older version of this branch

rljacob commented:
merged to next again

rljacob commented:
The test finally ran on gcp12 but gave an error during archiving.   I think its a CIME error.  CIME issue opened at https://github.com/ESMCI/cime/issues/4447

rljacob commented:
@sbrus89 where have you run this test successfully?   SMS_D_Ln3.TL319_EC30to60E2r2_wQU225EC30to60E2r2.GMPAS-JRA1p5-WW3.ww3-jra_1958.

sbrus89 commented:
@rljacob, I should have mentioned that I've always run into the archiving error you just made an issue for.

rljacob commented:
I'm trying to run on Chrysalis to reproduce and got an error at runtime: 
```
1279: forrtl: severe (194): Run-Time Check Failure. The variable 'w3initmd_mp_w3init_$OPENED' is being used in '/gpfs/fs1/home/jacob/E3SM2/components/ww3/src/WW3/model/tmp/w3initmd.F90(545,15)' without being defined
1279: Image              PC                Routine            Line        Source
1279: e3sm.exe           0000000004F7493E  w3initmd_mp_w3ini         545  w3initmd.F90
1279: e3sm.exe           0000000004B7DE4C  wav_comp_mct_mp_w         796  wav_comp_mct.F90
```

sbrus89 commented:
Thanks, I'll look into it.

rljacob commented:
I got the same error on Anvil intel.  Which machine compiler where you able to run and get the archive error?

sbrus89 commented:
I fixed the undefined variable issue in WW3 and updated the submodule. I was able to get this to run up to the archive error on Anvil (intel) and Chrysalis (intel, gnu). I'm seeing a run fail with gnu on Anvil and I'm looking into it.

sbrus89 commented:
@rljacob, I am seeing lots of these errors in the e3sm log for the Anvil gnu test:

```
1279: MPI startup(): Could not import some environment variables. Intel MPI process pinning will not be used.
1279:                Possible reason: Using the Slurm srun command. In this case, Slurm pinning will be used.
1279: MPIR_pmi_virtualization(): MPI startup(): PMI calls are forwarded to /usr/lib64/libpmi2.so
```
are there any known issues related to this?

sbrus89 commented:
The run step passes on Anvil with gnu if `--mpilib=openmpi` is specified in the `create_test` line.
